lib/include/jxl/codestream_header.h

```diff
@@ -72,6 +72,15 @@ typedef struct {
   uint32_t ysize;
 } JxlPreviewHeader;
 
+/** The intrinsic size header */
+typedef struct {
+  /** Intrinsic width in pixels */
+  uint32_t xsize;
+
+  /** Intrinsic height in pixels */
+  uint32_t ysize;
+} JxlIntrinsicSizeHeader;
+
```

Comment:
I would either just inline this struct in the BasicInfo struct (`intrinsic_xsize`, `intrinsic_ysize`), or define a generic `JxlSize` struct that is used anywhere where we need to specify image dimensions.
Currently we have a somewhat inconsistent mix of things:

- image dimensions are inlined: JxlBasicInfo.xsize/ysize
- frame dimensions are inlined in JxlFrameHeader.layer_info
- preview dimensions are in JxlPreviewHeader (which contains nothing else than an xsize/ysize so maybe JxlPreviewSize would have been a better name)
- intrinsic dimensions are in JxlIntrinsicSizeHeader

I think maybe we should just always inline things, i.e. have JxlBasicInfo.preview_xsize/ysize and JxlBasicInfo.intrinsic_xsize/ysize instead of having extra structs for it. A struct with just two fields is not really worth it imo.

Reply:
Now I just add two fileds to `JxlBasicInfo`, thereby using only the 8 bytes you provisioned.